File: civitasJava/src/juegoTexto/Controlador.java
/**
 * @author Yesenia González Dávila
 * @author Esther García Gallego Grupo B.3
 * @file Controlador.java
 */
package juegoTexto;

import civitas.CivitasJuego;
import civitas.Diario;
import civitas.OperacionesJuego;
import civitas.Respuestas;
import civitas.Jugador;
import civitas.Casilla;
import civitas.OperacionInmobiliaria;
import civitas.GestionesInmobiliarias;
import civitas.SalidasCarcel;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;

import java.util.ArrayList;

public class Controlador {

    private CivitasJuego juego;
    private VistaTextual vista;

    Controlador(CivitasJuego _juego, VistaTextual _vista) {
        juego = _juego;
        vista = _vista;
    }

    void juega() {
        boolean end = false;
        OperacionesJuego operacion;
        Respuestas respuesta;

        System.out.println("\n");

        vista.setCivitasJuego(juego);

        while (!end) {
            vista.actualizarVista();
            vista.pausa();

            operacion = juego.siguientePaso();
            vista.mostrarSiguienteOperacion(operacion);

            if (operacion != OperacionesJuego.PASAR_TURNO) {
                vista.mostrarEventos();
            }

            if (juego.finalDelJuego()) {
                end = true;
                ArrayList<Jugador> rank = new ArrayList(juego.ranking());

                for (int i = 0; i < rank.size(); i++) {
                    rank.get(i).toString();
                }
            } else {
                switch (operacion) {

                    case COMPRAR:
                        respuesta = vista.comprar();

                        if (respuesta == Respuestas.SI) {
                            juego.comprar();
                        }
                        juego.siguientePasoCompletado(operacion);
                        break;

                    case GESTIONAR:
                        vista.gestionar();
                        GestionesInmobiliarias gest = GestionesInmobiliarias.values()[vista.getGestion()];
                        int ip = vista.getPropiedad();

                        OperacionInmobiliaria operacionInm = new OperacionInmobiliaria(ip, gest);

                        switch (operacionInm.getGestion()) {
                            case VENDER:
                                juego.vender(ip);
                                break;
                            case HIPOTECAR:
                                juego.hipotecar(ip);
                                break;
                            case CANCELAR_HIPOTECA:
                                juego.cancelarHipoteca(ip);
                                break;
                            case CONSTRUIR_CASA:
                                juego.construirCasa(ip);
                                break;
                            case CONSTRUIR_HOTEL:
                                juego.construirHotel(ip);
                                break;
                            case TERMINAR:
                                juego.siguientePasoCompletado(operacion);
                                break;
                        }
                        break;

                    case SALIR_CARCEL:

                        SalidasCarcel salida = vista.salirCarcel();
                        if (salida == SalidasCarcel.PAGANDO) {
                            juego.salirCarcelPagando();
                        } else {
                            juego.salirCarcelTirando();
                        }

                        juego.siguientePasoCompletado(operacion);
                        break;
                }
            }

        }
    }
}

File: civitasRuby/pkg/civitasRuby-0.0.1/lib/JuegoTexto/controlador.rb
#encoding:utf-8
=begin
Authors: Esther García Gallego
         Yesenia González Dávila
         Grupo B3
=end
require_relative "../civitas/civitas_juego.rb"
require_relative "./vista_textual"
require_relative '../civitas/Casilla'
require_relative '../civitas/operacion_inmobiliaria'
require_relative '../civitas/gestiones_inmobiliarias'
require_relative '../civitas/salidas_carcel'

module Civitas
  class Controlador
    
    public # -----------------------------------------------------------
    
    def initialize(_juego, _vista)
      @juego = _juego
      @vista = _vista
    end
    
    def juega
      final = false
      
      puts "\n"
      @vista.setCivitasJuego(@juego)
      
      while (!final)
        @vista.actualizarVista
        @vista.pausa

        operacion = @juego.siguientePaso
        @vista.mostrarSiguienteOperacion(operacion)
          
        if operacion != Civitas::Operaciones_juego::PASAR_TURNO
          @vista.mostrarEventos
        end
          
        if @juego.finalDelJuego
          final = true
          #rank = Array.new(@juego.ranking.size) # Creamos el array

          rank = @juego.ranking         

          puts "\n ---------------------------- "
          puts   "     RANKING DE JUGADORES    "
          puts   " ----------------------------\n\n"
          
          for i in 0..1 # Mostramos los jugadores en orden
            puts rank[i].to_s
          end
          
          puts "\n ----- FIN DEL JUEGO -----\n"
          
        else
          case operacion                                    
            
          when Civitas::Operaciones_juego::COMPRAR
            respuesta = @vista.comprar
            if respuesta == Civitas::Respuestas::SI
              @juego.comprar
            end
            @juego.siguientePasoCompletado(operacion)
                  
          when Civitas::Operaciones_juego::GESTIONAR   
            @vista.gestionar
            
            #lista = [GestionesInmobiliarias::VENDER, GestionesInmobiliarias::HIPOTECAR, GestionesInmobiliarias::CANCELAR_HIPOTECA, GestionesInmobiliarias::CONSTRUIR_CASA, GestionesInmobiliarias::CONSTRUIR_HOTEL, GestionesInmobiliarias::TERMINAR]
            gest = Civitas::GestionesInmobiliarias::Lista_Gestiones[@vista.iGestion]
            #gest = lista[@vista.iGestion]
            ip = @vista.iPropiedad
                  
            operacionInm = OperacionInmobiliaria.new(ip,gest)
                  
            case operacionInm.gestion
            when Civitas::GestionesInmobiliarias::VENDER
              @juego.vender(ip)
              
            when Civitas::GestionesInmobiliarias::HIPOTECAR
              @juego.hipotecar(ip)
              
            when Civitas::GestionesInmobiliarias::CANCELAR_HIPOTECA
              @juego.cancelarHipoteca(ip)
              
            when Civitas::GestionesInmobiliarias::CONSTRUIR_CASA
              @juego.construirCasa(ip)
              
            when Civitas::GestionesInmobiliarias::CONSTRUIR_HOTEL
              @juego.construirHotel(ip)
              
            when Civitas::GestionesInmobiliarias::TERMINAR
              @juego.siguientePasoCompletado(operacion)
            end
                  
          when Civitas::Operaciones_juego::SALIR_CARCEL
            salida = @vista.salirCarcel
            
            if salida == Civitas::SalidasCarcel::PAGANDO
              @juego.salirCarcelPagando
            else
              @juego.salirCarcelTirando 
            end
                  
            @juego.siguientePasoCompletado(operacion)
            
          end
        
        end
      end
    
    end
  end
end

